src/spark_rapids_ml/knn.py

```diff
@@ -148,6 +142,36 @@ def getIdCol(self) -> str:
         )
         return col_name
 
+    def ensureIdCol(self, df: DataFrame) -> DataFrame:
+        """
+        Ensure an id column exists in the input dataframe. Add the column if not exists.
+        """
+        if not self.isDefined("id_col") and self.getIdCol() in df.columns:
```

Comment:
This is a bit unconventional w.r.t. Spark params.  Normally, if `isDefined("id_col")` was false, then `self.getIdCol()` should raise an error.  You _could_ try using `isSet("id_col")` to check if the user has defined the param, and then use `self.getOrDefault("id_col")` to get a default value of "id" (set via `_setDefault()`), so this would now be:
```
if not self.isSet("id_col") and self.getOrDefault("id_col") in df.columns:
```

Reply:
Nice finding. Thanks!